### PR TITLE
Add async timeout event to span

### LINKED ISSUE #35: Timeout details never actually used
https://github.com/opentracing-contrib/java-web-servlet-filter/blob/master/opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java#L109-L112

### PR DESCRIPTION
**Pull request by pavolloffay**
Fixes #35 

Signed-off-by: Pavol Loffay <ploffay@redhat.com>

### CODE REVIEW

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java [hunk 1]
```diff
@@ -181,20 +181,40 @@ public void doFilter(ServletRequest servletRequest, ServletResponse servletRespo
                             .addListener(new AsyncListener() {
                         @Override
                         public void onComplete(AsyncEvent event) throws IOException {
-                            try (Scope asyncScope = tracer.scopeManager().activate(scope.span(), true)) {
```
Comment (pavolloffay) [inline opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java:184]: I have removed activation for decorator, span is passed there explicitly. We don't need to create additional objects

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java [hunk 2]
```diff
@@ -196,6 +204,36 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
...
 
+    public static class AsyncTimeoutServlet extends HttpServlet {
+
+        @Override
+        public void doGet(final HttpServletRequest request, final HttpServletResponse response)
+            throws ServletException, IOException {
+
+            // avoid retries on timeout
+            if (request.getAttribute("timedOut") != null) {
+                return;
```
Comment (pavolloffay) [inline opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java:214]: @jpkrohling @objectiser if I don't do this the servlet is called multiple times. Any idea why?

**Review by jpkrohling — APPROVED** [review]
I haven't tested, but LGTM

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java [hunk 3]
```diff
@@ -46,6 +49,7 @@
     protected Server jettyServer;
     protected MockTracer mockTracer;
 
+
```
Comment (objectiser) [inline opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java]: nit: unnecessary additional line

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java (hunk 2 above)
Comment (objectiser) [reply opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java:214]: no idea - but seems strange, it is an auto retry mechanism?

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java [hunk 4]
```diff
@@ -219,6 +224,40 @@ public void testAsync() throws IOException {
         Assert.assertEquals(204, mockSpan.tags().get(Tags.HTTP_STATUS.getKey()));
         Assert.assertEquals("java-web-servlet", mockSpan.tags().get(Tags.COMPONENT.getKey()));
     }
+    @Test
```
Comment (objectiser) [inline opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/TracingFilterTest.java]: nit: line above

**Review by objectiser — APPROVED** [review]
Only some minor format issues.

File: opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java (hunk 2 above)
Comment (pavolloffay) [reply opentracing-web-servlet-filter/src/test/java/io/opentracing/contrib/web/servlet/filter/AbstractJettyTest.java:214]: I don't know whether it is a jetty thing or servlet spec

File: opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java [hunk 5]
```diff
@@ -105,11 +105,11 @@ public void onError(HttpServletRequest httpServletRequest, HttpServletResponse h
         public void onTimeout(HttpServletRequest httpServletRequest, HttpServletResponse httpServletResponse,
                               long timeout, Span span) {
             Tags.ERROR.set(span, Boolean.TRUE);
-
             Map<String, Object> timeoutLogs = new HashMap<>();
             timeoutLogs.put("event", Tags.ERROR.getKey());
```
Comment (pavolloffay) [inline opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:109]: Actually do we want to classify this as an error?

Comment (pavolloffay) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:109]: Maybe `event: timeout` and timeout value are enough.

Comment (objectiser) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:109]: Sounds like a discussion for the OT community, as it should be consistent across all instrumentations?

Comment (pavolloffay) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:109]: https://github.com/opentracing/specification/blob/master/semantic_conventions.md#log-fields-table in the event there is "timed out"

Comment (pavolloffay) [reply opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/ServletFilterSpanDecorator.java:109]: Please comment on #38

**Pull request merged**